test/unknown_command.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+set -o errexit
+set -o nounset
+
+TMP="$(mktemp -d)"
+clean() { rm -rf "$TMP"; }
+trap clean EXIT
+
+"$1" unknown_command 1> "$TMP/stdout" 2> "$TMP/stderr" || exit_code=$?
+
+if [ -z "${exit_code:-}" ] || [ "$exit_code" -eq 0 ]; then
```

Comment:
If you look at other files, we put `then` in its own line. In general, please try to copy the current style as much as possible, otherwise we end up with files that have their own style